### ISSUE TITLE
Submit GillesPy2 1.5.6

### ISSUE THREAD

**Issue by jonrkarr** [biosimulators/Biosimulators#81]
---
id: gillespy2
version: 1.5.6
specificationsUrl: https://raw.githubusercontent.com/biosimulators/Biosimulators_GillesPy2/e5db0434dc18125b8147f1335cb3c85a5e16e45b/biosimulators.json
specificationsPatch:
  version: 1.5.6
  image:
    url: ghcr.io/biosimulators/biosimulators_gillespy2/gillespy2:1.5.6
validateImage: true
commitSimulator: true

---

**Comment by biosimulators-daemon** [comment]
Thank you @jonrkarr for your submission to the BioSimulators simulator validation/submission system!

The BioSimulators validator bot is validating the specifications of your simulator, validating your Docker image and committing your simulator to the BioSimulators registry. 

We will discuss any concerns with your submission in this issue.

A complete log of your simulator submission job is available [here](https://github.com/biosimulators/Biosimulators/actions/runs/460670792).

**Comment by biosimulators-daemon** [comment]
The specifications of your simulator is valid!

**Comment by biosimulators-daemon** [comment]
## Summary of tests

* Executed 35 test cases

* Passed 10 test cases:
  * `cli.CliDescribesSupportedEnvironmentVariablesInline`
  * `cli.CliDisplaysHelpInline`
  * `cli.CliDisplaysVersionInformationInline`
  * `docker_image.DeclaresSupportedEnvironmentVariables`
  * `docker_image.DefaultUserIsRoot`
  * `docker_image.HasBioContainersLabels`
  * `docker_image.HasOciLabels`
  * `published_project.SimulatorCanExecutePublishedProject:sbml-core/Edelstein-Biol-Cybern-1996-Nicotinic-excitation`
  * `sedml.SimulatorSupportsAlgorithmParameters`
  * `sedml.SimulatorSupportsUniformTimeCoursesWithNonZeroInitialTimes`

* Failed 16 test cases:
  * `combine_archive.WhenACombineArchiveHasAMasterFileSimulatorOnlyExecutesThisFile`
  * `combine_archive.WhenACombineArchiveHasNoMasterFileSimulatorExecutesAllSedDocuments`
  * `exec_status_report.SimulatorReportsTheStatusOfTheExecutionOfCombineArchives`
  * `published_project.SimulatorCanExecutePublishedProject:sbml-core/Ciliberto-J-Cell-Biol-2003-morphogenesis-checkpoint`
  * `results_report.SimulatorGeneratesReportsOfSimultionResults`
  * `sedml.SimulatorProducesLinear2DPlots`
  * `sedml.SimulatorProducesLinear3DPlots`
  * `sedml.SimulatorProducesLogarithmic2DPlots`
  * `sedml.SimulatorProducesLogarithmic3DPlots`
  * `sedml.SimulatorProducesMultiplePlots`
  * `sedml.SimulatorProducesReportsWithCuratedNumberOfDimensions`
  * `sedml.SimulatorSupportsModelAttributeChanges`
  * `sedml.SimulatorSupportsModelsSimulationsTasksDataGeneratorsAndReports`
  * `sedml.SimulatorSupportsMultipleReportsPerSedDocument`
  * `sedml.SimulatorSupportsMultipleTasksPerSedDocument`
  * `sedml.SimulatorSupportsUniformTimeCoursesWithNonZeroOutputStartTimes`

* Skipped 9 test cases:
  * `published_project.SimulatorCanExecutePublishedProject:bngl/Dolan-PLoS-Comput-Biol-2015-NHEJ`
  * `published_project.SimulatorCanExecutePublishedProject:bngl/test-bngl`
  * `published_project.SimulatorCanExecutePublishedProject:sbml-core/Caravagna-J-Theor-Biol-2010-tumor-suppressive-oscillations`
  * `published_project.SimulatorCanExecutePublishedProject:sbml-core/Parmar-BMC-Syst-Biol-2017-iron-distribution`
  * `published_project.SimulatorCanExecutePublishedProject:sbml-core/Szymanska-J-Theor-Biol-2009-HSP-synthesis`
  * `published_project.SimulatorCanExecutePublishedProject:sbml-core/Tomida-EMBO-J-2003-NFAT-translocation`
  * `published_project.SimulatorCanExecutePublishedProject:sbml-core/Varusai-Sci-Rep-2018-mTOR-signaling-LSODA-LSODAR-SBML`
  * `published_project.SimulatorCanExecutePublishedProject:sbml-core/Vilar-PNAS-2002-minimal-circardian-clock`
  * `published_project.SimulatorCanExecutePublishedProject:sbml-fbc/Escherichia-coli-core-metabolism`


## Failures

### `combine_archive.WhenACombineArchiveHasAMasterFileSimulatorOnlyExecutesThisFile` (2.7 s)

  Test that when a COMBINE/OMEX archive defines a (single) master file, the simulator only executes this file.

  Exception:

  ```
  The image 'ghcr.io/biosimulators/biosimulators_gillespy2/gillespy2:1.5.6' could not execute the archive:
  
    Unknown error
  ```

  Log:

  ```
  Found 1 SED-ML documents with 1 models, 1 simulations, 1 tasks, 1 reports, and 0 plots:
    simulation_1.sedml:
      Tasks (1):
        simulation_1
      Reports (1):
        simulation_1: 20 data sets
  
  Executing SED-ML file 0: simulation_1.sedml
    Found 1 tasks
      simulation_1
    Executing task 1: simulation_1
  argument 'solver=<class 'gillespy2.solvers.numpy.ssa_solver.NumPySSASolver'>' to run() failed.  Reason Given: name 'Clog' is not defined
  ```
### `combine_archive.WhenACombineArchiveHasNoMasterFileSimulatorExecutesAllSedDocuments` (2.8 s)

  Test that when a COMBINE/OMEX archive does not have a defined master file, the simulator executes all SED-ML files.

  Exception:

  ```
  The image 'ghcr.io/biosimulators/biosimulators_gillespy2/gillespy2:1.5.6' could not execute the archive:
  
    Unknown error
  ```

  Log:

  ```
  Found 2 SED-ML documents with 2 models, 2 simulations, 2 tasks, 2 reports, and 0 plots:
    __copy__simulation_1.sedml:
      Tasks (1):
        simulation_1
      Reports (1):
        simulation_1: 20 data sets
    simulation_1.sedml:
      Tasks (1):
        simulation_1
      Reports (1):
        simulation_1: 20 data sets
  
  Executing SED-ML file 0: __copy__simulation_1.sedml
    Found 1 tasks
      simulation_1
    Executing task 1: simulation_1
  argument 'solver=<class 'gillespy2.solvers.numpy.ssa_solver.NumPySSASolver'>' to run() failed.  Reason Given: name 'Clog' is not defined
  ```
### `exec_status_report.SimulatorReportsTheStatusOfTheExecutionOfCombineArchives` (2.6 s)

  Test that when a COMBINE/OMEX archive defines a (single) master file, the simulator only executes this file.

  Exception:

  ```
  The image 'ghcr.io/biosimulators/biosimulators_gillespy2/gillespy2:1.5.6' could not execute the archive:
  
    Unknown error
  ```

  Log:

  ```
  Found 1 SED-ML documents with 1 models, 1 simulations, 1 tasks, 1 reports, and 0 plots:
    simulation_1.sedml:
      Tasks (1):
        simulation_1
      Reports (1):
        simulation_1: 20 data sets
  
  Executing SED-ML file 0: simulation_1.sedml
    Found 1 tasks
      simulation_1
    Executing task 1: simulation_1
  argument 'solver=<class 'gillespy2.solvers.numpy.ssa_solver.NumPySSASolver'>' to run() failed.  Reason Given: name 'Clog' is not defined
  ```
### `published_project.SimulatorCanExecutePublishedProject:sbml-core/Ciliberto-J-Cell-Biol-2003-morphogenesis-checkpoint` (2.6 s)

  Required model formats and simulation algorithms for SED tasks:
  
  * Format: [`format_2585`](https://www.ebi.ac.uk/ols/ontologies/edam/terms?iri=http%3A%2F%2Fedamontology.org%2Fformat_2585)
    Algorithm: [`KISAO_0000029`](https://www.ebi.ac.uk/ols/ontologies/kisao/terms?iri=http%3A%2F%2Fwww.biomodels.net%2Fkisao%2FKISAO%23KISAO_0000029)

  Exception:

  ```
  The image 'ghcr.io/biosimulators/biosimulators_gillespy2/gillespy2:1.5.6' could not execute the archive:
  
    Unknown error
  ```

  Log:

  ```
  Found 1 SED-ML documents with 1 models, 1 simulations, 1 tasks, 1 reports, and 0 plots:
    simulation_1.sedml:
      Tasks (1):
        simulation_1
      Reports (1):
        simulation_1: 20 data sets
  
  Executing SED-ML file 0: simulation_1.sedml
    Found 1 tasks
      simulation_1
    Executing task 1: simulation_1
  argument 'solver=<class 'gillespy2.solvers.numpy.ssa_solver.NumPySSASolver'>' to run() failed.  Reason Given: name 'Clog' is not defined
  ```
### `results_report.SimulatorGeneratesReportsOfSimultionResults` (2.6 s)

  Test that when a COMBINE/OMEX archive defines a (single) master file, the simulator only executes this file.

  Exception:

  ```
  The image 'ghcr.io/biosimulators/biosimulators_gillespy2/gillespy2:1.5.6' could not execute the archive:
  
    Unknown error
  ```

  Log:

  ```
  Found 1 SED-ML documents with 1 models, 1 simulations, 1 tasks, 1 reports, and 0 plots:
    simulation_1.sedml:
      Tasks (1):
        simulation_1
      Reports (1):
        simulation_1: 20 data sets
  
  Executing SED-ML file 0: simulation_1.sedml
    Found 1 tasks
      simulation_1
    Executing task 1: simulation_1
  argument 'solver=<class 'gillespy2.solvers.numpy.ssa_solver.NumPySSASolver'>' to run() failed.  Reason Given: name 'Clog' is not defined
  ```
### `sedml.SimulatorProducesLinear2DPlots` (2.6 s)

  Test that a simulator produces linear 2D plots

  Exception:

  ```
  The image 'ghcr.io/biosimulators/biosimulators_gillespy2/gillespy2:1.5.6' could not execute the archive:
  
    Unknown error
  ```

  Log:

  ```
  Found 1 SED-ML documents with 1 models, 1 simulations, 1 tasks, 0 reports, and 1 plots:
    simulation_1.sedml:
      Tasks (1):
        simulation_1
      Plots (1):
        plot_0: 20 curves
  
  Executing SED-ML file 0: simulation_1.sedml
    Found 1 tasks
      simulation_1
    Executing task 1: simulation_1
  argument 'solver=<class 'gillespy2.solvers.numpy.ssa_solver.NumPySSASolver'>' to run() failed.  Reason Given: name 'Clog' is not defined
  ```
### `sedml.SimulatorProducesLinear3DPlots` (2.7 s)

  Test that a simulator produces linear 3D plots

  Exception:

  ```
  The image 'ghcr.io/biosimulators/biosimulators_gillespy2/gillespy2:1.5.6' could not execute the archive:
  
    Unknown error
  ```

  Log:

  ```
  Found 1 SED-ML documents with 1 models, 1 simulations, 1 tasks, 0 reports, and 1 plots:
    simulation_1.sedml:
      Tasks (1):
        simulation_1
      Plots (1):
        plot_0: 20 surfaces
  
  Executing SED-ML file 0: simulation_1.sedml
    Found 1 tasks
      simulation_1
    Executing task 1: simulation_1
  argument 'solver=<class 'gillespy2.solvers.numpy.ssa_solver.NumPySSASolver'>' to run() failed.  Reason Given: name 'Clog' is not defined
  ```
### `sedml.SimulatorProducesLogarithmic2DPlots` (2.7 s)

  Test that a simulator produces logarithmic 2D plots

  Exception:

  ```
  The image 'ghcr.io/biosimulators/biosimulators_gillespy2/gillespy2:1.5.6' could not execute the archive:
  
    Unknown error
  ```

  Log:

  ```
  Found 1 SED-ML documents with 1 models, 1 simulations, 1 tasks, 0 reports, and 1 plots:
    simulation_1.sedml:
      Tasks (1):
        simulation_1
      Plots (1):
        plot_0: 20 curves
  
  Executing SED-ML file 0: simulation_1.sedml
    Found 1 tasks
      simulation_1
    Executing task 1: simulation_1
  argument 'solver=<class 'gillespy2.solvers.numpy.ssa_solver.NumPySSASolver'>' to run() failed.  Reason Given: name 'Clog' is not defined
  ```
### `sedml.SimulatorProducesLogarithmic3DPlots` (2.9 s)

  Test that a simulator produces logarithmic 3D plots

  Exception:

  ```
  The image 'ghcr.io/biosimulators/biosimulators_gillespy2/gillespy2:1.5.6' could not execute the archive:
  
    Unknown error
  ```

  Log:

  ```
  Found 1 SED-ML documents with 1 models, 1 simulations, 1 tasks, 0 reports, and 1 plots:
    simulation_1.sedml:
      Tasks (1):
        simulation_1
      Plots (1):
        plot_0: 20 surfaces
  
  Executing SED-ML file 0: simulation_1.sedml
    Found 1 tasks
      simulation_1
    Executing task 1: simulation_1
  argument 'solver=<class 'gillespy2.solvers.numpy.ssa_solver.NumPySSASolver'>' to run() failed.  Reason Given: name 'Clog' is not defined
  ```
### `sedml.SimulatorProducesMultiplePlots` (2.7 s)

  Test that a simulator produces multiple plots

  Exception:

  ```
  The image 'ghcr.io/biosimulators/biosimulators_gillespy2/gillespy2:1.5.6' could not execute the archive:
  
    Unknown error
  ```

  Log:

  ```
  Found 1 SED-ML documents with 1 models, 1 simulations, 1 tasks, 0 reports, and 2 plots:
    simulation_1.sedml:
      Tasks (1):
        simulation_1
      Plots (2):
        plot_0: 10 curves
        plot_1: 10 curves
  
  Executing SED-ML file 0: simulation_1.sedml
    Found 1 tasks
      simulation_1
    Executing task 1: simulation_1
  argument 'solver=<class 'gillespy2.solvers.numpy.ssa_solver.NumPySSASolver'>' to run() failed.  Reason Given: name 'Clog' is not defined
  ```
### `sedml.SimulatorProducesReportsWithCuratedNumberOfDimensions` (2.6 s)

  Test that that the curated number of output dimensions matches the actual number of output dimensions

  Exception:

  ```
  The image 'ghcr.io/biosimulators/biosimulators_gillespy2/gillespy2:1.5.6' could not execute the archive:
  
    Unknown error
  ```

  Log:

  ```
  Found 1 SED-ML documents with 1 models, 1 simulations, 1 tasks, 1 reports, and 0 plots:
    simulation_1.sedml:
      Tasks (1):
        simulation_1
      Reports (1):
        simulation_1: 20 data sets
  
  Executing SED-ML file 0: simulation_1.sedml
    Found 1 tasks
      simulation_1
    Executing task 1: simulation_1
  argument 'solver=<class 'gillespy2.solvers.numpy.ssa_solver.NumPySSASolver'>' to run() failed.  Reason Given: name 'Clog' is not defined
  ```
### `sedml.SimulatorSupportsModelAttributeChanges` (2.7 s)

  Test that a simulator supports changes to the attributes of model elements

  Exception:

  ```
  The image 'ghcr.io/biosimulators/biosimulators_gillespy2/gillespy2:1.5.6' could not execute the archive:
  
    Unknown error
  ```

  Log:

  ```
  Found 1 SED-ML documents with 1 models, 1 simulations, 1 tasks, 1 reports, and 0 plots:
    simulation_1.sedml:
      Tasks (1):
        simulation_1
      Reports (1):
        simulation_1: 20 data sets
  
  Executing SED-ML file 0: simulation_1.sedml
    Found 1 tasks
      simulation_1
    Executing task 1: simulation_1
  argument 'solver=<class 'gillespy2.solvers.numpy.ssa_solver.NumPySSASolver'>' to run() failed.  Reason Given: name 'Clog' is not defined
  ```
### `sedml.SimulatorSupportsModelsSimulationsTasksDataGeneratorsAndReports` (2.6 s)

  Test that a simulator supports the core elements of SED: models, simulations, tasks, data generators for individual variables, and reports

  Exception:

  ```
  The image 'ghcr.io/biosimulators/biosimulators_gillespy2/gillespy2:1.5.6' could not execute the archive:
  
    Unknown error
  ```

  Log:

  ```
  Found 1 SED-ML documents with 1 models, 1 simulations, 1 tasks, 1 reports, and 0 plots:
    simulation_1.sedml:
      Tasks (1):
        simulation_1
      Reports (1):
        simulation_1: 20 data sets
  
  Executing SED-ML file 0: simulation_1.sedml
    Found 1 tasks
      simulation_1
    Executing task 1: simulation_1
  argument 'solver=<class 'gillespy2.solvers.numpy.ssa_solver.NumPySSASolver'>' to run() failed.  Reason Given: name 'Clog' is not defined
  ```
### `sedml.SimulatorSupportsMultipleReportsPerSedDocument` (2.6 s)

  Test that a simulator supports multiple reports per SED document

  Exception:

  ```
  The image 'ghcr.io/biosimulators/biosimulators_gillespy2/gillespy2:1.5.6' could not execute the archive:
  
    Unknown error
  ```

  Log:

  ```
  Found 1 SED-ML documents with 1 models, 1 simulations, 1 tasks, 2 reports, and 0 plots:
    simulation_1.sedml:
      Tasks (1):
        simulation_1
      Reports (2):
        report_1: 10 data sets
        report_2: 11 data sets
  
  Executing SED-ML file 0: simulation_1.sedml
    Found 1 tasks
      simulation_1
    Executing task 1: simulation_1
  argument 'solver=<class 'gillespy2.solvers.numpy.ssa_solver.NumPySSASolver'>' to run() failed.  Reason Given: name 'Clog' is not defined
  ```
### `sedml.SimulatorSupportsMultipleTasksPerSedDocument` (2.7 s)

  Test that a simulator supports multiple tasks per SED document

  Exception:

  ```
  The image 'ghcr.io/biosimulators/biosimulators_gillespy2/gillespy2:1.5.6' could not execute the archive:
  
    Unknown error
  ```

  Log:

  ```
  Found 1 SED-ML documents with 1 models, 1 simulations, 2 tasks, 2 reports, and 0 plots:
    simulation_1.sedml:
      Tasks (2):
        simulation_1
        simulation_1__test_suite_copy
      Reports (2):
        simulation_1: 20 data sets
        simulation_1__test_suite_copy: 20 data sets
  
  Executing SED-ML file 0: simulation_1.sedml
    Found 2 tasks
      simulation_1
      simulation_1__test_suite_copy
    Executing task 1: simulation_1
  argument 'solver=<class 'gillespy2.solvers.numpy.ssa_solver.NumPySSASolver'>' to run() failed.  Reason Given: name 'Clog' is not defined
  ```
### `sedml.SimulatorSupportsUniformTimeCoursesWithNonZeroOutputStartTimes` (2.7 s)

  Test that a simulator supports time courses with non-zero output start times

  Exception:

  ```
  The image 'ghcr.io/biosimulators/biosimulators_gillespy2/gillespy2:1.5.6' could not execute the archive:
  
    Unknown error
  ```

  Log:

  ```
  Found 1 SED-ML documents with 1 models, 1 simulations, 1 tasks, 1 reports, and 0 plots:
    simulation_1.sedml:
      Tasks (1):
        simulation_1
      Reports (1):
        simulation_1: 20 data sets
  
  Executing SED-ML file 0: simulation_1.sedml
    Found 1 tasks
      simulation_1
    Executing task 1: simulation_1
  argument 'solver=<class 'gillespy2.solvers.numpy.ssa_solver.NumPySSASolver'>' to run() failed.  Reason Given: name 'Clog' is not defined
  ```


## Warnings

### `cli.CliDescribesSupportedEnvironmentVariablesInline` (10.0 s)

  Test that the inline help for a command-line interface describes the environment variables that the simulator supports.

  Warnings:

  ```
  The inline help for a command-line interface for a simulation tool should describe the environment variables that the simulation tool supports.
  
  The command-line interface does not describe the following standard environment variables recognized by BioSimulators:
    - 'ALGORITHM_SUBSTITUTION_POLICY'
  
  If the simulation tool implements these variables, they should be described in the inline help for its command-line interface.
  
  Note, support for these environment variables is optional. Simulation tools are not required to support these variables.
  ```

  Log:

  ```
  
  ```
### `cli.CliDisplaysVersionInformationInline` (3.7 s)

  Test that a command-line interface provides version information inline.

  Warnings:

  ```
  Command-line interface should support the `-v` option for displaying version information inline.
  
  The command-line interface displayed the following when executed with `-v`:
  
    GillesPy2: 1.5.6, CLI: 0.1.16, Python: 3.7.9, OS: Linux debian 10.7 5.4.0-1032-azure, Machine: x86_64
    
  ```

  ```
  Command-line interface should support the `--version` option for displaying version information inline.
  
  The command-line interface displayed the following when executed with `--version`:
  
    GillesPy2: 1.5.6, CLI: 0.1.16, Python: 3.7.9, OS: Linux debian 10.7 5.4.0-1032-azure, Machine: x86_64
    
  ```

  Log:

  ```
  
  ```
### `docker_image.DeclaresSupportedEnvironmentVariables` (2.3 s)

  Test if a Docker image declares the environment variables that is supports

  Warnings:

  ```
  Docker images for simulation tools should declare the environment variables that they support.
  
  The Docker image does not declare the following standard environment variables recognized by BioSimulators:
    - 'ALGORITHM_SUBSTITUTION_POLICY'
  
  If the simulation tool implements these variables, they should be declared in the Dockerfile for the Docker image for the simulator.
  
  Note, support for these environment variables is optional. Simulation tools are not required to support these variables.
  ```

  Log:

  ```
  
  ```
### `docker_image.HasBioContainersLabels` (1.1 s)

  Test that a Docker image has BioContainers labels with metadata about the image

  Warnings:

  ```
  The Docker image should have the following BioContainers labels:
    extra.identifiers.biotools
  ```

  Log:

  ```
  
  ```
### `sedml.SimulatorSupportsUniformTimeCoursesWithNonZeroInitialTimes` (2.6 s)

  Test that a simulator supports multiple time courses with non-zero initial times

  Warnings:

  ```
  The image 'ghcr.io/biosimulators/biosimulators_gillespy2/gillespy2:1.5.6' could not execute the archive:
  
    Unknown error
  ```

  Log:

  ```
  Found 1 SED-ML documents with 1 models, 1 simulations, 1 tasks, 1 reports, and 0 plots:
    simulation_1.sedml:
      Tasks (1):
        simulation_1
      Reports (1):
        simulation_1: 20 data sets
  
  Executing SED-ML file 0: simulation_1.sedml
    Found 1 tasks
      simulation_1
    Executing task 1: simulation_1
  Initial simulation time 50.0 is not supported. Initial time must be 0.
  ```

**Comment by biosimulators-daemon** [comment]
```diff
- After correcting your simulator, please edit the first block of this issue to re-initiate this validation.
```

**Comment by jonrkarr** [comment]
Rejecting GillesPy2 1.5.6 because it doesn't correctly handle species with ids that include "ln". This is fixed in GillesPy2 1.5.7.
